x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_edit_extension/endpoint_policy_edit_extension.test.tsx

```diff
@@ -53,55 +40,52 @@ describe('When displaying the EndpointPolicyEditExtension fleet UI extension', (
       );
   });
 
-  it('should show artifact cards', async () => {
-    const renderResult = render();
-
-    await waitFor(() => {
-      artifactCards.forEach((artifactCard) => {
-        expect(renderResult.getByTestId(artifactCard)).toBeTruthy();
-      });
-    });
+  afterEach(() => {
+    useUserPrivilegesMock.mockReturnValue(getUserPrivilegesMockDefaultValue());
   });
```

Comment:
I think this should be in the `beforeEach` hook, because now the first test case passes accidentally, because it receives a mocked return value that's okay, but it comes from somewhere else, it's not defined in this file.

Actually I think it is weird that the first test passes - the return value of `useUserPrivileges` should be `undefined` until we configure a mock return value. But it looks like it is configured somewhere, but I couldn't find where. Do you have any idea? Does it maybe come from the depths of `createFleetContextRendererMock()`?

Reply:
@gergoabraham ,

Yeah, some of our most used hooks are "auto-mocked" via Jest's `__mocks__` directory. The `useUserPrivileges()` hook is one of those that you don't have to explicitly set on initial load - its mocked to set all privileges to true here: 

https://github.com/elastic/kibana/blob/6efef0496077f4e61d49e4e43f75941ff3d98d9e/x-pack/plugins/security_solution/public/common/components/user_privileges/__mocks__/index.ts

Reply:
Oh, cool! That's good to know, thanks! And now I see that you are restoring exactly the same default implementation, great!